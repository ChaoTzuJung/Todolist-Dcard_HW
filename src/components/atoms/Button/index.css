.button {
	display: flex;
	justify-content: center;
	position: relative;
	width: 310px;
	height: 60px;
	padding: 16px 0;
	outline: none;
	border: none;
	transition: all 0.3s;

	& > div {
		display: flex;
		justify-content: center;
		align-items: center;
		text-align: center;
		font-weight: 400;
		font-size: 24px;
		color: #000;

		& > i {
			font-size: 24px;
			margin-right: 14px;
		}
	}

	&:hover {
		cursor: pointer;
	}

	&.save {
		background: #4a90e2;

		& > div,
		& i {
			color: #fff;
		}

		/* &:hover {
			background: var(--grayLight);
		}

		&:focus {
			border: 1px solid var(--accent);
		}

		&:active {
			border: 1px solid var(--accent);
			background: var(--pinkLight);
		} */
	}

	&.cancel {
		background: white;

		& > div,
		& i {
			color: #d0021b;
		}

		/* &:hover,
		&:focus,
		&:active {
			background: var(--accent);
		} */

		&.disabled {
			background: var(--white);
		}
	}

	&.upload {
		background: #c8c8c8;

		& > div,
		& i {
			color: #f2f2f2;
		}
	}

	&.disabled {
		background: solid 1px gray;

		& > div {
			color: #c8c8c8;

			& > i {
				color: #c8c8c8;
			}
		}
	}
}
