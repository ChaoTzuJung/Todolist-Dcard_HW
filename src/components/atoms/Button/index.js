import React from 'react';
import classnames from 'classnames';
import { Link } from 'react-router';

import styles from './index.css';

const Button = ({
	children,
	className,
	color,
	link,
	type = 'internal',
	disabled = false,
	onClick = () => {},
	...other
}) => {
	if (link) {
		if (type === 'external') {
			return (
				<a
					className={classnames(styles.button, className, {
						[styles.normal]: color === 'normal',
						[styles.save]: color === 'save',
						[styles.cancel]: color === 'cancel',
						[styles.upload]: color === 'upload',
						[styles.disabled]: disabled,
					})}
					href={link}
					target="_blank"
					rel="noopener noreferrer"
					onClick={e => {
						if (!disabled) {
							onClick(e);
						}
					}}
					{...other}
				>
					<div>{children}</div>
				</a>
			);
		}

		return (
			<Link
				className={classnames(styles.button, className, {
					[styles.normal]: color === 'normal',
					[styles.save]: color === 'save',
					[styles.cancel]: color === 'cancel',
					[styles.upload]: color === 'upload',
					[styles.disabled]: disabled,
				})}
				to={link}
				onClick={e => {
					if (!disabled) {
						onClick(e);
					}
				}}
				{...other}
			>
				<div>{children}</div>
			</Link>
		);
	}

	return (
		/* eslint-disable react/button-has-type */
		<button
			className={classnames(styles.button, className, {
				[styles.normal]: color === 'normal',
				[styles.save]: color === 'save',
				[styles.cancel]: color === 'cancel',
				[styles.upload]: color === 'upload',
				[styles.disabled]: disabled,
			})}
			onClick={e => {
				if (!disabled) {
					onClick(e);
				}
			}}
			{...other}
		>
			<div>{children}</div>
		</button>
	);
};

export default Button;
