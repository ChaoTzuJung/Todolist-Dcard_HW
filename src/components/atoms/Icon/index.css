.icon {
	color: #000000;

	&.light {
		color: #FFFFFF;
	}
}
