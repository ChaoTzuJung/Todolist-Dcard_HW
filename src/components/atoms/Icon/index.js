import React from 'react';
import classnames from 'classnames';

import styles from './index.css';

const Icon = ({
	children,
	className,
	light = false,
	size = 'md-24', // CSS rules for the standard material design sizing guidelines (.md-18 .md-24 .md-36 .md-48)
	...other
}) => (
	<i
		className={classnames(`material-icons ${size}`, styles.icon, className, {
			[styles.light]: light,
		})}
		{...other}
	>
		{children}
	</i>
);

export default Icon;
