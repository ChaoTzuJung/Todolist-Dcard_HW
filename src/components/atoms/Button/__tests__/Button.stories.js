import React from 'react';
import { withKnobs, select, text, boolean } from '@storybook/addon-knobs/react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';

import Button from 'components/atoms/Button';
import Icon from 'components/atoms/Icon';

const stories = storiesOf('Commons|atoms/Button', module);

stories.addDecorator(withKnobs);

stories.add('__interactive', () => (
	<div style={{ border: '1px solid #000', margin: '60px', display: 'inline-block' }}>
		<Button
			color={select(
				'color theme',
				{ normal: 'normal', save: 'save', cancel: 'cancel', upload: 'upload' },
				'normal',
			)}
			disabled={boolean('Disabled', false)}
			onClick={action('onClick')}
		>
			{text('content', 'Add Task')}
		</Button>
	</div>
));

stories.add('with icon text', () => (
	<div style={{ border: '1px solid #000', margin: '60px', display: 'inline-block' }}>
		<Button>
			<Icon>{text('with icon', 'close')}</Icon>
			{text('text', 'Cancel')}
		</Button>
	</div>
));

stories.add('with internal link', () => (
	<div style={{ border: '1px solid #000', margin: '60px', display: 'inline-block' }}>
		<Button color="normal" link="/">
			{text('content', 'Go to home page')}
		</Button>
	</div>
));

stories.add('with external link', () => (
	<div style={{ border: '1px solid #000', margin: '60px', display: 'inline-block' }}>
		<Button color="normal" link="https://www.google.com/" type="external">
			{text('content', 'Link to google site')}
		</Button>
	</div>
));
