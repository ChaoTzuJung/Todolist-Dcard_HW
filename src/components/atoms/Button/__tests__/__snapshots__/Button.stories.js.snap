// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`__interactive 1`] = `
<Button
  color="normal"
  disabled={false}
  onClick={[Function]}
>
  <button
    className="button normal"
    onClick={[Function]}
  >
    <div>
      Add Task
    </div>
  </button>
</Button>
`;

exports[`with external link 1`] = `
<Button
  color="normal"
  link="https://www.google.com/"
  type="external"
>
  <a
    className="button normal"
    href="https://www.google.com/"
    onClick={[Function]}
    rel="noopener noreferrer"
    target="_blank"
  >
    <div>
      Link to google site
    </div>
  </a>
</Button>
`;

exports[`with icon text 1`] = `
<Button>
  <button
    className="button"
    onClick={[Function]}
  >
    <div>
      <Icon>
        <i
          className="icon material-icons"
        >
          close
        </i>
      </Icon>
      Cancel
    </div>
  </button>
</Button>
`;

exports[`with internal link 1`] = `
<Button
  color="normal"
  link="/"
>
  <Link
    className="button normal"
    onClick={[Function]}
    onlyActiveOnIndex={false}
    style={Object {}}
    to="/"
  >
    <a
      className="button normal"
      onClick={[Function]}
      style={Object {}}
    >
      <div>
        Go to home page
      </div>
    </a>
  </Link>
</Button>
`;
